### PR TITLE
Update docs for compat switch value

### PR DESCRIPTION
**Pull request by rynowak**
https://github.com/aspnet/Mvc/issues/8595

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Core/CompatibilityVersion.cs [hunk 1]
```diff
@@ -66,7 +66,15 @@ public enum CompatibilityVersion
         /// <remarks>
         /// ASP.NET Core MVC 2.2 introduces compatibility switches for the following:
         /// <list type="bullet">
+        ///     <item><description><c>ApiBehaviouOptions.SuppressMapClientErrors</c></description></item>
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/CompatibilityVersion.cs]: While you're here, could you update the 2.1 list too:

``
MvcOptions.AllowCombiningAuthorizeFilters
                    MvcOptions.AllowBindingHeaderValuesToNonStringModelTypes
                    MvcOptions.AllowValidatingTopLevelNodes
                    MvcOptions.InputFormatterExceptionPolicy
                    MvcOptions.SuppressBindingUndefinedValueToEnumType
```

**Comment by pranavkm** [comment]
Looks like this broke the functional tests. Can you make sure all the changes are correct before updating this PR:

```
 The active test run was aborted. Reason: 
2018-10-12T21:54:18.6180772Z   [xUnit.net 00:00:00.72] Microsoft.AspNetCore.Mvc.FunctionalTests: Skipping test case with duplicate ID 'cfda34017a23dca0236f913c460b4762216dc179' ('Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_MultipleRouteAttributes_WithMultipleHttpAttributes_RespectsConstraints(url: "http://localhost/api/v1/Maps/5", method: "PATCH")' and 'Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_MultipleRouteAttributes_WithMultipleHttpAttributes_RespectsConstraints(url: "http://localhost/api/v1/Maps/5", method: "PATCH")')
2018-10-12T21:54:18.6182831Z   [xUnit.net 00:00:00.72] Microsoft.AspNetCore.Mvc.FunctionalTests: Skipping test case with duplicate ID 'f548cda654b9f2eeb803ecda8f2a5585a9a7d7ea' ('Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_MultipleRouteAttributes_WithMultipleHttpAttributes_RespectsConstraints(url: "http://localhost/api/v2/Maps/5", method: "PATCH")' and 'Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_MultipleRouteAttributes_WithMultipleHttpAttributes_RespectsConstraints(url: "http://localhost/api/v2/Maps/5", method: "PATCH")')
2018-10-12T21:54:18.6271885Z   [xUnit.net 00:00:00.72] Microsoft.AspNetCore.Mvc.FunctionalTests: Skipping test case with duplicate ID '15b634906f1e8ff9bdf3c71b3ac473354b42e7fd' ('Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_MultipleRouteAttributes_WithMultipleHttpAttributes_RespectsConstraints(url: "http://localhost/api/v1/Maps/PartialUpdate/5", method: "PUT")' and 'Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_MultipleRouteAttributes_WithMultipleHttpAttributes_RespectsConstraints(url: "http://localhost/api/v1/Maps/PartialUpdate/5", method: "PUT")')
2018-10-12T21:54:18.6273378Z   [xUnit.net 00:00:00.72] Microsoft.AspNetCore.Mvc.FunctionalTests: Skipping test case with duplicate ID '63eb01f480385ad8005a7c8a438b867095fc5323' ('Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_MultipleRouteAttributes_WithMultipleHttpAttributes_RespectsConstraints(url: "http://localhost/api/v2/Maps/PartialUpdate/5", method: "PUT")' and 'Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_MultipleRouteAttributes_WithMultipleHttpAttributes_RespectsConstraints(url: "http://localhost/api/v2/Maps/PartialUpdate/5", method: "PUT")')
2018-10-12T21:54:18.6274505Z   [xUnit.net 00:00:00.72] Microsoft.AspNetCore.Mvc.FunctionalTests: Skipping test case with duplicate ID '591a20ed706a5ae7859340b9b4999d58669854ca' ('Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_RejectsRequestsWithWrongMethods_InRoutesWithoutExtraTemplateSegmentsOnTheAction(method: "Post", url: "/Friends")' and 'Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_RejectsRequestsWithWrongMethods_InRoutesWithoutExtraTemplateSegmentsOnTheAction(method: "Post", url: "/Friends")')
2018-10-12T21:54:18.6275781Z   [xUnit.net 00:00:00.72] Microsoft.AspNetCore.Mvc.FunctionalTests: Skipping test case with duplicate ID 'fe19d87be7ce67d5a6b8714f2e499bf4037a9ddd' ('Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_RejectsRequestsWithWrongMethods_InRoutesWithoutExtraTemplateSegmentsOnTheAction(method: "Put", url: "/Friends")' and 'Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_RejectsRequestsWithWrongMethods_InRoutesWithoutExtraTemplateSegmentsOnTheAction(method: "Put", url: "/Friends")')
2018-10-12T21:54:18.6277689Z   [xUnit.net 00:00:00.72] Microsoft.AspNetCore.Mvc.FunctionalTests: Skipping test case with duplicate ID 'fcc4f1f6ee12c97955a92a8b88da0c7da9c3e401' ('Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_RejectsRequestsWithWrongMethods_InRoutesWithoutExtraTemplateSegmentsOnTheAction(method: "Patch", url: "/Friends")' and 'Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_RejectsRequestsWithWrongMethods_InRoutesWithoutExtraTemplateSegmentsOnTheAction(method: "Patch", url: "/Friends")')
2018-10-12T21:54:18.6279131Z   [xUnit.net 00:00:00.72] Microsoft.AspNetCore.Mvc.FunctionalTests: Skipping test case with duplicate ID 'c17ab370ce1da0de1294dd7c226f39f0cd0e779d' ('Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_RejectsRequestsWithWrongMethods_InRoutesWithoutExtraTemplateSegmentsOnTheAction(method: "Options", url: "/Friends")' and 'Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_RejectsRequestsWithWrongMethods_InRoutesWithoutExtraTemplateSegmentsOnTheAction(method: "Options", url: "/Friends")')
2018-10-12T21:54:18.6280663Z   [xUnit.net 00:00:00.72] Microsoft.AspNetCore.Mvc.FunctionalTests: Skipping test case with duplicate ID 'a3a65bce75dba4f8bd2a0d43585423e65db7cf40' ('Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_RejectsRequestsWithWrongMethods_InRoutesWithoutExtraTemplateSegmentsOnTheAction(method: "Head", url: "/Friends")' and 'Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRoutedAction_RejectsRequestsWithWrongMethods_InRoutesWithoutExtraTemplateSegmentsOnTheAction(method: "Head", url: "/Friends")')
2018-10-12T21:54:18.6281889Z   [xUnit.net 00:00:00.72] Microsoft.AspNetCore.Mvc.FunctionalTests: Skipping test case with duplicate ID '952bb54f4fb8f1d72213289cab17b638381d6747' ('Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRouting_MixedAcceptVerbsAndRoute_Unreachable(path: "/Bank/Deposit", verb: "GET")' and 'Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRouting_MixedAcceptVerbsAndRoute_Unreachable(path: "/Bank/Deposit", verb: "GET")')
2018-10-12T21:54:18.6282901Z   [xUnit.net 00:00:00.72] Microsoft.AspNetCore.Mvc.FunctionalTests: Skipping test case with duplicate ID '9f34cc884248b1d2bcdac4a51ac593a155428cf0' ('Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRouting_MixedAcceptVerbsAndRoute_Unreachable(path: "/Bank/Deposit/5", verb: "DELETE")' and 'Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRouting_MixedAcceptVerbsAndRoute_Unreachable(path: "/Bank/Deposit/5", verb: "DELETE")')
2018-10-12T21:54:18.6283849Z   [xUnit.net 00:00:00.72] Microsoft.AspNetCore.Mvc.FunctionalTests: Skipping test case with duplicate ID '20d6dfebae2472f336295d9f8ea9d7cb5f135173' ('Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRouting_MixedAcceptVerbsAndRoute_Unreachable(path: "/Bank/Withdraw/5", verb: "GET")' and 'Microsoft.AspNetCore.Mvc.FunctionalTests.RoutingEndpointRoutingTest.AttributeRouting_MixedAcceptVerbsAndRoute_Unreachable(path: "/Bank/Withdraw/5", verb: "GET")')
2018-10-12T21:54:53.3654814Z   Results File: /home/vsts/work/1/s/artifacts/logs/UnitTests-netcoreapp2.2-20181012.10.trx
2018-10-12T21:54:53.3662961Z   
2018-10-12T21:54:53.3671081Z   Total tests: Unknown. Passed: 12282. Failed: 0. Skipped: 2.
2018-10-12T21:54:53.3673346Z   Test Run Aborted.
2018-10-12T21:54:53.3673628Z   Test execution time: 1.3669 Minutes
```

**Comment by rynowak** [comment]
```
2018-10-12T21:54:53.3671081Z   Total tests: Unknown. Passed: 12282. Failed: 0. Skipped: 2.
2018-10-12T21:54:53.3673346Z   Test Run Aborted.
2018-10-12T21:54:53.3673628Z   Test execution time: 1.3669 Minutes
```

**Comment by rynowak** [comment]
The CI failure here was a timeout of the run - waiting too long for an agent, so I'm ignoring it.

**Pull request merged**